util/codec/codec.go

```diff
@@ -52,7 +53,18 @@ func encode(b []byte, vals []types.Datum, comparable bool) ([]byte, error) {
 			b = encodeBytes(b, val.GetBytes(), comparable)
 		case types.KindMysqlTime:
 			b = append(b, uintFlag)
-			v, err := val.GetMysqlTime().ToPackedUint()
+			t := val.GetMysqlTime()
+			// Encoding timestamp need to consider timezone.
+			// if it's not in UTC, transform to UTC first.
+			if t.Type == mysql.TypeTimestamp && t.TimeZone != time.UTC && !t.IsZero() {
```

Comment:
Good question. @XuHuaiyu 
Convert timezone need to call Go's library, if `t.IsZero()`, convert t to Go's time would get `1970-0-0`, the data will be wrong.
That would not a wrong behavior, as MySQL also take zero time specially.